### PR TITLE
Add Assistant Vision example to README

### PR DESCRIPTION
**Pull request by alexrudall**
## All Submissions:

* [ ] Have you followed the guidelines in our [Contributing document](../blob/main/CONTRIBUTING.md)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1072,6 +1072,71 @@ run_id = response['id']
...
+# Upload image as a file
+file_id = client.files.upload(
+  parameters: {
+    file: "path/to/example.png",
+    purpose: "assistants"
+  }
+)["id"]
+
+# Create assistant
+assistant_id = client.assistants.create(
```
Comment (mtantawy) [inline README.md:1097]: First of all, thank you for documenting this and providing an example, that's really helpful!

If I could only make a change here that would be somehow communicating the fact that creating an assistant should be a one-time action, a user won't need to recreate the assistant for every attempt to describe or read an image, that's what threads are for

Comment (alexrudall) [reply README.md:1097]: To be honest I treat them as pretty throwaway as well :D 

What about `# Create assistant (or grab an existing one`

Comment (mtantawy) [reply README.md:1097]: It's true, they come for free, I like to keep them unique for organization purposes anyway

> To be honest I treat them as pretty throwaway as well :D
> 
> What about `# Create assistant (or grab an existing one)`

That sounds good, there is always room for clarifying more but that's probably out of scope of your example here

Comment (alexrudall) [reply README.md:1097]: Alrighty, I tweaked it a bit more [here](https://github.com/alexrudall/ruby-openai?tab=readme-ov-file#vision-in-a-thread)

Thanks for the suggestion!

**Pull request merged**